### PR TITLE
separate prow url and batch merge in submit-queue

### PR DESCRIPTION
**Pull request by BenTheElder**
Fixes https://github.com/kubernetes/test-infra/issues/3624

### CODE REVIEW

File: mungegithub/mungers/submit-queue.go [hunk 1]
```diff
@@ -258,10 +258,11 @@ type SubmitQueue struct {
 
 	features *features.Features
 
-	mergeLock   sync.Mutex // acquired when attempting to merge a specific PR
-	BatchURL    string
-	ContextURL  string
-	batchStatus submitQueueBatchStatus
+	mergeLock    sync.Mutex // acquired when attempting to merge a specific PR
+	ProwDataURL  string
```
Comment (rmmh) [inline mungegithub/mungers/submit-queue.go]: just call it ProwURL, and comment that it's the JSON jobs result page.

(Data means almost nothing without more context)

File: mungegithub/mungers/submit-queue.go [hunk 2]
```diff
@@ -551,7 +551,8 @@ func (sq *SubmitQueue) RegisterOptions(opts *options.Options) {
 	// If you create a StringSliceVar you may wish to check out 'cleanStringSlice()'
 	opts.RegisterString(&sq.Metadata.HistoryUrl, "history-url", "", "URL to access the submit-queue instance's health history.")
 	opts.RegisterString(&sq.Metadata.ChartUrl, "chart-url", "", "URL to access the submit-queue instance's health charts.")
-	opts.RegisterString(&sq.BatchURL, "batch-url", "", "Prow data.json URL to read batch results")
+	opts.RegisterString(&sq.ProwDataURL, "prow-data-url", "", "Prow data.json URL to read batch results")
+	opts.RegisterBool(&sq.BatchEnabled, "batch-enabled", false, "Batch merge")
```
Comment (rmmh) [inline mungegithub/mungers/submit-queue.go]: change help to "Do batch merges (requires prow/splice coordination)."

File: mungegithub/mungers/submit-queue.go [hunk 3]
```diff
@@ -486,9 +487,8 @@ func (sq *SubmitQueue) internalInitialize(config *github.Config, features *featu
 
 	go sq.handleGithubE2EAndMerge()
 	go sq.updateGoogleE2ELoop()
-	if sq.BatchURL != "" {
+	if sq.BatchEnabled {
```
Comment (rmmh) [inline mungegithub/mungers/submit-queue.go]: can you add a check earlier in this function to return an error if BatchEnabled but ProwURL is empty?

**Comment by BenTheElder** [comment]
Yes, I will make those changes in a moment.

**Comment by BenTheElder** [comment]
The error on BatchEnabled and empty ProwURL may still be nit-worthy but the rest should be fixed.

File: mungegithub/mungers/submit-queue.go (hunk 3 above)
Comment (BenTheElder) [inline mungegithub/mungers/submit-queue.go]: Done.

**Comment by kargakis** [comment]
Do you guys forsee the submit queue using ProwJobs for something else other than batch merges?

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/test-infra/issues/3624#issuecomment-317050129

Yes, I foresee generally tracking job metrics more for the submit-queue dash. We have something for nonblocking e2e now but not really for other job types.

**Pull request merged**